onnxscript/optimizer/_constant_folding.py

```diff
@@ -817,9 +834,16 @@ def visit_node(self, node: ir.Node, root: ir.Graph | ir.Function):
             self.replace_node(node, replacement, root)
 
     def visit_graph(self, graph: ir.Graph) -> None:
+        self._state.push_initializer_inputs()
+        for input in graph.inputs:
+            if input.const_value is not None:
+                self._state.add_initializer_input(input)
```

Comment:
Possible to check the graph.initializers.values() field for initializers?

Reply:
I don't see the value in doing that. This is more efficient. And it should be equivalent currently. Further, the real goal here is to stop treating `.const_value` as a constant-value if it is a graph input (so this condition reflects exactly the intended purpose). (Actually, we can omit the if-condition completely, but this is an optimization to track just inputs with a const-value instead of all inputs.)

Reply:
Oh I see. initializers will have non-input values. Makes sense